core/utils/xml.ts

```diff
@@ -14,40 +14,69 @@
 import * as goog from '../../closure/goog/goog.js';
 goog.declareModuleId('Blockly.utils.xml');
 
+import * as deprecation from './deprecation.js';
+
 
 /**
- * Namespace for Blockly's XML.
+ * Injected dependencies.  By default these are just (and have the
+ * same types as) the corresponding DOM Window properties, but the
+ * Node.js wrapper for Blockly (see scripts/package/node/core.js)
+ * calls injectDependencies to supply implementations from the jsdom
+ * package instead.
+ */
+let {document, DOMParser, XMLSerializer} = globalThis;
+
+/**
+ * Inject dependencies.  Used by the Node.js wrapper for Blockly (see
+ * scripts/package/node/core.js) to supply implementations from the
+ * jsdom package instead.  While they may be set individually, it is
+ * normally the case that all three should be supplied and sourced
+ * from the same JSDOM instance.
  *
- * @alias Blockly.utils.xml.NAME_SPACE
+ * @param dependencies Options object contianing dependencies to set.
  */
-export const NAME_SPACE = 'https://developers.google.com/blockly/xml';
+export function injectDependencies(dependencies: {
+  document?: Document,
+  DOMParser?: typeof DOMParser,
+  XMLSerializer?: typeof XMLSerializer,
+}) {
+  ({
+    // Default to existing value if option not supplied.
+    document = document,
+    DOMParser = DOMParser,
+    XMLSerializer = XMLSerializer,
+  } = dependencies);
+}
 
 /**
- * The Document object to use.  By default this is just document, but
- * the Node.js build of Blockly (see scripts/package/node/core.js)
- * calls setDocument to supply a Document implementation from the
- * jsdom package instead.
+ * Namespace for Blockly's XML.
+ *
+ * @alias Blockly.utils.xml.NAME_SPACE
  */
-let xmlDocument: Document = globalThis['document'];
+export const NAME_SPACE = 'https://developers.google.com/blockly/xml';
 
 /**
  * Get the document object to use for XML serialization.
  *
  * @returns The document object.
+ * @deprecated No longer provided by Blockly.
  * @alias Blockly.utils.xml.getDocument
  */
 export function getDocument(): Document {
-  return xmlDocument;
+  deprecation.warn('Blockly.utils.xml.getDocument', 'version 9', 'version 10');
```

Comment:
The long version is: "use the value you passed to `injectDependencies`, or if you didn't do that and are on node.js then just create a new `JSDOM` instance and use its `.window.document`, or if you're in a browser then just use `window.document` aka `globalThis.document` aka `document."

I'm not sure how to summarise that concisely, though, and I'm not sure it's worth bothering, because AFAICT this getter was created (as `document`) in #3126 for the sole purpose of being monkey-patchable by the node.js version of `core.js`.  It should have been deprecated when the only call to it (in this module) was removed in #5461, and I'm having trouble envisioning which "people" might have been using it and for what for…  Why would someone want to get a `Document` instance _from_ Blockly??

Reply:
"what are you doing and can you not?"